gst/tensor_merge/gsttensormerge.c

```diff
@@ -460,52 +496,53 @@ gst_tensor_merge_collect_buffer (GstTensorMerge * tensor_merge,
     walk = g_slist_next (walk);
 
     GstBuffer *buf = NULL;
-    buf = gst_collect_pads_pop (tensor_merge->collect, data);
-
-    if (GST_IS_BUFFER (buf)) {
-      if (GST_BUFFER_PTS_IS_VALID (buf)) {
-        if (data->segment.format == GST_FORMAT_TIME)
-          pad->pts_timestamp =
-              gst_segment_to_running_time (&data->segment, GST_FORMAT_TIME,
-              GST_BUFFER_PTS (buf));
-      } else {
-        pad->pts_timestamp = GST_CLOCK_TIME_NONE;
-      }
-      if (GST_BUFFER_DTS_IS_VALID (buf)) {
-        if (data->segment.format == GST_FORMAT_TIME)
-          pad->dts_timestamp =
-              gst_segment_to_running_time (&data->segment, GST_FORMAT_TIME,
-              GST_BUFFER_DTS (buf));
-      } else {
-        pad->dts_timestamp = GST_CLOCK_TIME_NONE;
-      }
 
-      mem = gst_buffer_get_memory (buf, 0);
-      gst_buffer_append_memory (tensors_buf, mem);
+    buf = gst_collect_pads_peek (tensor_merge->collect, data);
 
-      if (gst_tensor_merge_compare_pads (tensor_merge, bestpad, pad) > 0) {
-        bestpad = pad;
-        *pts_time = bestpad->pts_timestamp;
-        *dts_time = bestpad->dts_timestamp;
+    if (buf == NULL && pad->buffer == NULL)
+      return TRUE;
+
+    if (buf != NULL) {
+      if (GST_BUFFER_PTS (buf) < tensor_merge->current_time) {
+        buf = gst_collect_pads_pop (tensor_merge->collect, data);
+        pad->buffer = buf;
+        silent_debug ("Fame Dropped : %lu", GST_BUFFER_PTS (buf));
+        tensor_merge->need_buffer = TRUE;
+        gst_buffer_unref (tensors_buf);
+        return FALSE;
       }
 
-      gst_buffer_unref (buf);
+      if (pad->buffer != NULL
+          && ABS (tensor_merge->current_time - GST_BUFFER_PTS (pad->buffer) <
+              ABS (tensor_merge->current_time - GST_BUFFER_PTS (buf)))) {
+        buf = pad->buffer;
```

Comment:
gst_buffer_unref (buf) before changing to pad->buffer

Reply:
Fixed.